core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2627,6 +2627,37 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  // Create an instance of the class with the given name, possibly initializing it with our conf
+  def instantiateClass[T](className: String, conf: SparkConf, isDriver: Boolean): T = {
```

Comment:
Make this private ?

Reply:
Fixed.

Reply:
Oh, we cannot really make it private since it is used in SparkEnv.scala:314.
```
[error] /home/runner/work/spark/spark/core/src/main/scala/org/apache/spark/SparkEnv.scala:314:32: method instantiateClass in object Utils cannot be accessed in object org.apache.spark.util.Utils
[error]     val shuffleManager = Utils.instantiateClass[ShuffleManager](shuffleMgrClass, conf, isDriver)
```

Reply:
This method is highly specific to a Serializer instance, right? then that can be reflected in the name and types

Reply:
Yes. The new code moves `instantiateClassFromConf` and `instantiateClass` from SparkEnv into Utils, in order to let `isPushBasedShuffleEnabled` to leverage them for instantiating serializer instances.

Reply:
@rmcyang Can you rename the method as suggested by @srowen ? It is not a general instantiation of class; but for a class which exposes one of the supported constructor signature.
Essentially it is specific to either `Serializer` or `ShuffleManager`.

Reply:
The `instantiateClassFromConf` could be made private, but I don't think this `instantiateClass` method can:

it's also used to instantiate user-configurable ShuffleManagers: https://github.com/apache/spark/pull/34158/files#diff-7f3f6f8b1c65a0a1f984854d7cb9d97e848c07b96278513a519bde658331a37eL340-R314

Reply:
Since `instantiateClass` is used for either `Serializer` or `ShuffleManager`, I'm thinking renaming it as `instantiateClassForSerializerOrShuffleManager`. Let me know for any better name in your minds. @srowen @mridulm

Reply:
> The instantiateClassFromConf could be made private

@JoshRosen Can `instantiateClassFromConf` really made private? It's also used to initiate Serializer instance in SparkEnv:
https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkEnv.scala#L301

Reply:
Whoops, I overlooked that: it looks like neither of these methods can be made private.